.github/workflows/ci.yml

```
@@ -689,6 +721,34 @@ jobs:
           restore-keys: packages-cachebust-3-
           fail-on-cache-miss: true
 
+      - uses: actions/cache@v3
+        name: Cargo Cache
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ./target/
+            ./firmware/examples/target/
+          key: cargo-cache-cachebust-2-${{ hashFiles('Cargo.lock') }}
+
+      - name: Build Firmware Examples
+        run: cargo build --release
+        working-directory: firmware/examples/
+
+      - name: Show project files
+        run: |
+          find . -type f \
+               -not -path './.git/*' \
+            -a -not -path './firmware/examples/target/*' \
+            -a -not -path './clash-vexriscv/clash-vexriscv/build_out_dir/*' \
+            -a -not -path './clash-vexriscv/clash-vexriscv/example-cpu/target/*' \
+            -a -not -path './clash-vexriscv/clash-vexriscv/example-cpu/project/target/*' \
+            -a -not -path './_build/*' \
+            -a -not -path './target/*' \
+            -a -not -path './dist-newstyle/*' \
```

```suggestion
      - name: Show project files
        run: |
          git ls-files
```

? Or is this not what you're after?

---

agreed